umbral/pre.py

```diff
@@ -24,6 +24,10 @@
 class GenericUmbralError(Exception):
     pass
 
+class UmbralCorrectnessError(GenericUmbralError):
+    def __init__(self, message, offending_cfrags):
+        super().__init__(message)
+        self.offending_cfrags = offending_cfrags
```

Comment:
Is there a need to capture these here even if we handle them ~L485?

Reply:
The idea is to include them in the exception, so in the application (i.e., the KMS), Bob could know which cfrags were incorrect and act consequently.